psycopg/psycopg/_typeinfo.py

```diff
@@ -15,13 +15,15 @@
 from .abc import AdaptContext
 from .rows import dict_row
 
+
 if TYPE_CHECKING:
     from .connection import Connection
     from .connection_async import AsyncConnection
-    from .sql import Identifier
+    from .sql import Composed, Identifier, SQL
 
 T = TypeVar("T", bound="TypeInfo")
 RegistryKey: TypeAlias = Union[str, int, Tuple[type, int]]
+_AnyConn: TypeAlias = Union["Connection[Any]", "AsyncConnection[Any]"]
```

Comment:
This signature was already used here, @CaselIT only made a declaration for it. I seem to remember that there is a reason to use `Union[sync, async]` - I can't remember which one, but likely something related to something missing on BaseConnection.

Maybe `Union["Connection[Any]", "AsyncConnection[Any]"]` is important enough to deserve a place in `psycopg.abc`. I've been thinking to add it as `AnyConnection`, but there's ambiguity to what the `Any` refers to (the type, or the asyncness?)

Reply:
I agree the already-there `Union` is needed because `BaseConnection` is not enough (namely, in `fetch()`, the `transaction()` method is used). But in newly added methods, only `.info` is used, so `BaseConnection` would be enough. Not a big deal though.

Reply:
I can move it to abc, but it's probably better to do this in a separate PR, since there are probably more cleanup opportunities.

I'm gonna restore the union, and use base connection in the other places so that's easier to cleanup later